### PR TITLE
Allow setting PREFIX in the build to dictate where to find directories

### PR DESCRIPTION
**Pull request by superm1**
This allows building many components without elevated permissions or needing to do it within a container.

### CODE REVIEW

**Comment by agners** [comment]
LGTM, I'll give it a try in my setup before merging.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,8 @@
-RELEASE_DIR = /build/release
+PREFIX:=$(PWD)
+RELEASE_DIR = $(PREFIX)/release
 
-BUILDROOT=/build/buildroot
-BUILDROOT_EXTERNAL=/build/buildroot-external
+BUILDROOT=$(PREFIX)/buildroot
```
Comment (agners) [inline Makefile:4]: `PREFIX` is used for target installation prefix in automake world, can we use something else, e.g. `BUILDDIR`?

**Review by agners — CHANGES_REQUESTED** [review]
Otherwise looks good, thanks!

**Comment by agners** [comment]
Actually, this seems not to work when using the `scripts/enter.sh` method as documented in `Documentation/getting_started_development.md`.

**Comment by superm1** [comment]
> Actually, this seems not to work when using the `scripts/enter.sh` method as documented in `Documentation/getting_started_development.md`.

I did double check that - 

```
~/src/home-assistant/operating-system$ git branch
  dev
* host-build
~/src/home-assistant/operating-system$ ./scripts/enter.sh 
Sending build context to Docker daemon  36.27MB
Step 1/7 : FROM debian:buster
 ---> 6d6b00c22231
Step 2/7 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> 3439c35ece3e
Step 3/7 : RUN apt-get update && apt-get install -y --no-install-recommends         apt-transport-https         ca-certificates         curl         gpg-agent         gpg         dirmngr         software-properties-common     && curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -     && add-apt-repository "deb https://download.docker.com/linux/debian $(lsb_release -cs) stable"     && apt-get update && apt-get install -y --no-install-recommends         docker-ce     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> ea166240db0e
Step 4/7 : RUN apt-get update && apt-get install -y --no-install-recommends         bash         bc         binutils         build-essential         bzip2         cpio         file         git         make         ncurses-dev         patch         perl         python         rsync         sudo         unzip         wget         qemu-utils         openssh-client         vim     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 5186f76f16a8
Step 5/7 : COPY scripts/entry.sh /usr/sbin/
 ---> Using cache
 ---> bc603b9df86b
Step 6/7 : ENTRYPOINT ["/usr/sbin/entry.sh"]
 ---> Using cache
 ---> ba1c6eb6bdcd
Step 7/7 : WORKDIR /build
 ---> Using cache
 ---> b6d22d33ad04
Successfully built b6d22d33ad04
Successfully tagged hassos:local
builder@2cf926cb4109:/build$ 
```

File: Makefile (hunk 1 above)
Comment (superm1) [reply Makefile:4]: Sure, good point.  I'll make this change, re-check it and push it again.

**Comment by agners** [comment]
> I did double check that -

Hm, I see, it works without arguments. I often use `./scripts/enter.sh make O=output_odroid_n2 odroid_n2`, which now seems to fail. Not sure why exactly, shouldn't make just merge those command line provided vars?

**Comment by agners** [comment]
What does work for me is using this in the `Makefile`:
```
ifeq ($(BUILDDIR),)
BUILDDIR := $(shell pwd)
endif
```

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,8 @@
-RELEASE_DIR = /build/release
+BUILDDIR:=$(PWD)
```
Comment (agners) [inline Makefile:1]:
```suggestion
ifeq ($(BUILDDIR),)
BUILDDIR := $(shell pwd)
endif
```

Comment (superm1) [reply Makefile]: The `ifeq` seems unnecessary, just shelling out is enough.

**Comment by superm1** [comment]
> What does work for me is using this in the Makefile:

I just pushed something like that but without the `ifeq` which seems to work for me.
```
$ ./scripts/enter.sh make help
Sending build context to Docker daemon  36.28MB
Step 1/7 : FROM debian:buster
 ---> 6d6b00c22231
Step 2/7 : SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 ---> Using cache
 ---> 3439c35ece3e
Step 3/7 : RUN apt-get update && apt-get install -y --no-install-recommends         apt-transport-https         ca-certificates         curl         gpg-agent         gpg         dirmngr         software-properties-common     && curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -     && add-apt-repository "deb https://download.docker.com/linux/debian $(lsb_release -cs) stable"     && apt-get update && apt-get install -y --no-install-recommends         docker-ce     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> ea166240db0e
Step 4/7 : RUN apt-get update && apt-get install -y --no-install-recommends         bash         bc         binutils         build-essential         bzip2         cpio         file         git         make         ncurses-dev         patch         perl         python         rsync         sudo         unzip         wget         qemu-utils         openssh-client         vim     && rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 5186f76f16a8
Step 5/7 : COPY scripts/entry.sh /usr/sbin/
 ---> Using cache
 ---> bc603b9df86b
Step 6/7 : ENTRYPOINT ["/usr/sbin/entry.sh"]
 ---> Using cache
 ---> ba1c6eb6bdcd
Step 7/7 : WORKDIR /build
 ---> Using cache
 ---> b6d22d33ad04
Successfully built b6d22d33ad04
Successfully tagged hassos:local
Supported targets: odroid_c4 intel_nuc ova rpi4_64 rpi2 odroid_c2 rpi4 tinker rpi odroid_xu4 rpi0_w rpi3_64 odroid_n2 rpi3
Run 'make <target>' to build a target image.
Run 'make all' to build all target images.
Run 'make clean' to clean the build output.
Run 'make <target>-config' to configure buildroot for a target.
```

Build targets only show up if the root is properly set up.

**Comment by agners** [comment]
> I just pushed something like that but without the `ifeq` which seems to work for me.

Works for me as well. Thanks!

**Pull request merged**